src/content/app/genome-browser/components/browser-sidebar-modal/modal-views/navigate-modal/NavigateLocationModal.tsx

```diff
@@ -0,0 +1,315 @@
+  updateBrowserSidebarModalForGenome
+} from 'src/content/app/genome-browser/state/browser-sidebar-modal/browserSidebarModalSlice';
+import type { ChrLocation } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSlice';
+
+import styles from './NavigateModal.scss';
+
+const NavigateLocationModal = () => {
+  const activeGenomeId = useAppSelector(getBrowserActiveGenomeId);
+
+  const dispatch = useAppDispatch();
+
+  const { data: genomeKaryotype = [] } = useGenomeKaryotypeQuery(
+    activeGenomeId as string
+  );
+  const { changeBrowserLocation } = useGenomeBrowser();
+
+  const [stickInput, setStickInput] = useState('');
+  const [locationStartInput, setLocationStartInput] = useState('');
+  const [locationEndInput, setLocationEndInput] = useState('');
+  const [regionInput, setRegionInput] = useState('');
+
+  const [coordInputsActive, setCoordInputsActive] = useState(false);
+  const [regionInputActive, setRegionInputActive] = useState(false);
+
+  const [locationStartErrorMessage, setLocationStartErrorMessage] = useState<
+    string | null
+  >(null);
+  const [locationEndErrorMessage, setLocationEndErrorMessage] = useState<
+    string | null
+  >(null);
+  const [regionErrorMessage, setRegionErrorMessage] = useState<string | null>(
+    null
+  );
+
+  const stickRef = useRef<HTMLDivElement>(null);
+  const locationStartRef = useRef<HTMLDivElement>(null);
+  const locationEndRef = useRef<HTMLDivElement>(null);
+  const regionRef = useRef<HTMLDivElement>(null);
+
+  const onCoordInputsFocus = () => {
+    setCoordInputsActive(true);
+    setRegionInputActive(false);
+  };
+
+  const onRegionInputFocus = () => {
+    setCoordInputsActive(false);
+    setRegionInputActive(true);
+  };
+
+  const onLocationStartChange = (event: FormEvent<HTMLInputElement>) => {
+    const value = event.currentTarget.value;
+    setLocationStartInput(value);
+    setLocationStartErrorMessage(null);
+  };
+
+  const onLocationEndChange = (event: FormEvent<HTMLInputElement>) => {
+    const value = event.currentTarget.value;
+    setLocationEndInput(value);
+    setLocationEndErrorMessage(null);
+  };
+
+  const onRegionChange = (event: FormEvent<HTMLInputElement>) => {
+    const value = event.currentTarget.value;
+    setRegionInput(value);
+    setRegionErrorMessage(null);
+  };
+
+  const handleSubmit = () => {
+    if (activeGenomeId) {
+      setRegionErrorMessage(null);
+
+      const newLocation = coordInputsActive
+        ? `${stickInput}:${locationStartInput}-${locationEndInput}`
+        : regionInput;
+
+      validateRegion({
+        regionInput: newLocation,
+        genomeId: activeGenomeId,
+        onSuccess: onValidationSuccess,
+        onError: onValidationError
+      });
+    }
+  };
+
+  const getKaryotypeOptions = () =>
+    genomeKaryotype.map(({ name }) => ({
+      value: name,
+      label: name,
+      isSelected: stickInput === name
+    }));
+
+  const updateStickInput = (value: string) => {
```

Comment:
`stick` is a term used inside the genome browser. In browser chrome, I think we call them regions.

Reply:
renamed this to `chrInput` as there is already a `regionInput`. Couldn't think of any better name. ✅

Reply:
> as there is already a regionInput

Shouldn't that `regionInput` be renamed to `locationInput`?

Reply:
✅